collector/processors/odigosroutingfilterprocessor/processor.go

```diff
@@ -0,0 +1,152 @@
+package odigosroutingfilterprocessor
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+)
+
+const (
+	k8sNamespaceNameAttr   = "k8s.namespace.name"
+	k8sDeploymentNameAttr  = "k8s.deployment.name"
+	k8sStatefulSetNameAttr = "k8s.statefulset.name"
+	k8sDaemonSetNameAttr   = "k8s.daemonset.name"
+	kindDeployment         = "Deployment"
+	kindStatefulSet        = "StatefulSet"
+	kindDaemonSet          = "DaemonSet"
+)
+
+type filterProcessor struct {
+	logger *zap.Logger
+	config *Config
+}
+
+func (fp *filterProcessor) processTraces(ctx context.Context, td ptrace.Traces) (ptrace.Traces, error) {
+	rspans := td.ResourceSpans()
+
+	for i := 0; i < rspans.Len(); i++ {
+		resourceSpan := rspans.At(i)
+		resourceAttributes := resourceSpan.Resource().Attributes()
+		ilSpans := resourceSpan.ScopeSpans()
+
+		for j := 0; j < ilSpans.Len(); j++ {
+			scopeSpan := ilSpans.At(j)
+			spans := scopeSpan.Spans()
+
+			spans.RemoveIf(func(span ptrace.Span) bool {
+				namespace, name, kind := extractResourceDetails(resourceAttributes)
+				return !fp.matches(name, namespace, kind)
+			})
+		}
```

Comment:
since you make the decision based on the resource attributes only, you can remove the spans (metrics/logs) iterations and simply remove or keep the entire scopeSpans